frontend/app/src/scenes/Home/scenes/AddCategory/index.tsx

```diff
@@ -10,35 +10,80 @@ import {
   Typography,
 } from 'antd';
 import mainStyles from '../../styles.css';
-import { AddCategoryStateProps } from './interfaces';
+import { AddCategoryStateProps, CategoryUrlParams } from './interfaces';
 import { RouteComponentProps, withRouter } from 'react-router';
-import axios, { AxiosResponse } from 'axios';
+import axios, { AxiosResponse, Method } from 'axios';
 import { Category } from '../../interfaces';
 import { handleApiError } from '../../../../services/util/errorHandler';
 
 const { Title, Text } = Typography;
 
 class AddCategory extends React.Component<
-  RouteComponentProps,
+  RouteComponentProps<CategoryUrlParams>,
   AddCategoryStateProps
 > {
-  constructor(props: any) {
+  categoryId: string;
+  constructor(props: RouteComponentProps<CategoryUrlParams>) {
     super(props);
+    this.categoryId = this.props.match.params.categoryId;
     this.state = {
       isLoading: false,
+      isTypeEdit: !!this.categoryId,
+      category: null,
     };
   }
 
-  onFinish = (values: any) => {
+  componentDidMount() {
+    if (this.state.isTypeEdit) {
+      this.fetchCategory();
+    }
+  }
+
+  fetchCategory = () => {
     this.setState({ isLoading: true });
     axios
-      .post(window.location.origin + '/core/academix/admin/categories', values)
+      .get(
+        window.location.origin + '/core/academix/categories/' + this.categoryId
+      )
+      .then((result: AxiosResponse<Category>) => {
+        if (result.status == 200) {
+          this.setState({ isLoading: false, category: result.data });
+        } else {
+          throw new Error();
+        }
+      })
+      .catch((error) => {
+        handleApiError(error, 'Something went wrong trying to load category');
+        this.setState({ isLoading: false });
+      });
+  };
+
+  onFinish = (values: any) => {
+    let statusCode: number, method: Method, url: string;
+    this.setState({ isLoading: true });
+    if (this.state.isTypeEdit) {
+      statusCode = 202;
```

Comment:
After my PR get's merged it will be 200.

Reply:
wrong issue number is linked in the PR

Reply:
Thanks, I will change it into 200 after your PR.

Reply:
let's add some comments to the code @Gravewalker666

Reply:
Sure!

Reply:
Resolved with commit [1628960](https://github.com/sef-global/sef-core/pull/81/commits/162896037fd9f9a8c5ea86a786ce67869de5071b